exporter/signalfxexporter/exporter.go

```diff
@@ -146,6 +143,29 @@ func newSignalFxExporter(
 	}, nil
 }
 
+func getHTTPClient(config *Config) (*http.Client, error) {
+	defaultMaxConns := 100
+	if config.MaxIdleConns == nil {
+		if config.MaxConnections != nil {
+			config.MaxIdleConns = config.MaxConnections
+		} else {
+			config.MaxIdleConns = &defaultMaxConns
+		}
+	}
+	if config.MaxIdleConnsPerHost == nil {
+		if config.MaxConnections != nil {
+			config.MaxIdleConnsPerHost = config.MaxConnections
+		} else {
+			config.MaxIdleConnsPerHost = &defaultMaxConns
+		}
+	}
+	// To avoid overriding client headers to request header storing access_token
+	// when access_token_passthrough is true.
+	config.HTTPClientSettings.Headers = map[string]string{}
```

Comment:
Appears we should respect these values and override any existing: https://github.com/open-telemetry/opentelemetry-collector/blob/e6f0e014c0ad280fc67557704a2015924caf76eb/config/confighttp/confighttp.go#L52

What problem are you experiencing with token passthrough?

Reply:
when access_token_passthrough is true, if we have SignalFx receiver and SignalFx exporter and an access token is provided to SignalFx receiver as header then that access token should be taken as an access token for the SignalFx exporter, irrespective of any user input of the exporter (as per current upstream implementation of the exporter). 
Now in SignalFx exporter config, if the user has provided access token in the header as “X-Sf-Token” and access_token_passthrough is true and if header(access token) is provided from SignalFx receiver then the access token passed from SignalFx receiver should be set as a header at the time of the client.Do(req) call, irrespective of any user input in the exporter.
Earlier we were calling the roundTrip function of net/http package where they were not overriding the request headers.  But when we use HTTPClientSettings, its [roundTrip function](https://github.com/open-telemetry/opentelemetry-collector/blob/51fcb65f02055638f223c6abd3d6d2d644c696db/config/confighttp/confighttp.go#L181) overrides the request headers and assigns the value to the user input access token. So at last the access token provided by the user will be passed, which is not expected behavior. Due to this, the following [unit test](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/38668ef013261b04e3de7b9d3a5bf0a100d7cbff/exporter/signalfxexporter/exporter_test.go#L302) was failing. 
So to resolve this we have set the headers of HTTPClientSettings to empty map. Still, if we want to retrieve our headers then they are available in headers config, so we have not dropped any of the headers.

Reply:
Ok, I think I understand the issue and why this is required for the client creation.

Reply:
If it looks good, could you please resolve this comment?

Reply:
I'm not able to resolve comments in this project

Reply:
okay, I do have access, I am resolving it.